android-pdf-viewer/src/main/java/com/infomaniak/lib/pdfview/scroll/DefaultScrollHandle.java

```diff
@@ -185,55 +255,132 @@ public void hide() {
         setVisibility(INVISIBLE);
     }
 
+    /**
+     * @param color color of the text handle
+     */
     public void setTextColor(int color) {
-        textView.setTextColor(color);
+        textColorResId = color;
     }
 
     /**
      * @param size text size in dp
      */
     public void setTextSize(int size) {
-        textView.setTextSize(TypedValue.COMPLEX_UNIT_DIP, size);
+        textSize = size;
+    }
+
+    /**
+     * @param handleBackgroundDrawable drawable to set as a background
+     */
+    public void setPageHandleBackground(Drawable handleBackgroundDrawable) {
+        this.handleBackgroundDrawable = handleBackgroundDrawable;
+    }
+
+    /**
+     * Use a custom view as the handle. Note that this view should have a TextView with a specific tag,
+     * "pageIndicator", in order to display the current page
+     *
+     * @param handleView view to set as the handle
+     */
+    public void setPageHandleView(View handleView) {
+        this.handleView = handleView;
+    }
+
+    /**
+     * @param handleWidth  width of the handle
+     * @param handleHeight width of the handle
+     */
+    public void setHandleSize(int handleWidth, int handleHeight) {
+        this.handleWidth = handleWidth;
+        this.handleHeight = handleHeight;
+    }
+
+    /**
+     * @param paddingLeft   left padding of the handle
+     * @param paddingTop    top padding of the handle
+     * @param paddingRight  right padding of the handle
+     * @param paddingBottom bottom padding of the handle
+     */
+    public void setHandlePaddings(int paddingLeft, int paddingTop, int paddingRight, int paddingBottom) {
+        handlePaddingLeft = paddingLeft;
+        handlePaddingTop = paddingTop;
+        handlePaddingRight = paddingRight;
+        handlePaddingBottom = paddingBottom;
+    }
+
+    /**
+     * @param hideHandleDelayMillis delay in milliseconds to hide the handle after scrolling the PDF
+     */
+    public void setHideHandleDelay(int hideHandleDelayMillis) {
+        this.hideHandleDelayMillis = hideHandleDelayMillis;
     }
 
     private boolean isPDFViewReady() {
         return pdfView != null && pdfView.getPageCount() > 0 && !pdfView.documentFitsView();
     }
 
+    private View getTouchedView(MotionEvent event) {
+        int x = Math.round(event.getX());
+        int y = Math.round(event.getY());
+        for (int i = 0; i < getChildCount(); i++) {
+            View child = getChildAt(i);
+            if (x > child.getLeft() && x < child.getRight() && y > child.getTop() && y < child.getBottom()) {
+                return child;
+            }
+        }
+        return null;
+    }
+
+    private boolean shouldIgnoreTouch(MotionEvent event) {
+        View touchedView = getTouchedView(event);
+        if (isTouched) {
+            return false;
+        } else if (touchedView != null) {
+            Object tag = touchedView.getTag();
+            return tag != null && !tag.toString().equals("rootHandle");
```

Comment:
What's best between asking users to set a tag "rootHandle" to the view that will act as a the handle or asking the user to implement a function that will return the handle view at all time?